datafusion/common/src/stats.rs

```diff
@@ -436,6 +492,8 @@ pub struct ColumnStatistics {
     pub max_value: Precision<ScalarValue>,
     /// Minimum value of column
     pub min_value: Precision<ScalarValue>,
+    /// Sum value of a column
+    pub sum_value: Precision<ScalarValue>,
```

Comment:
As I think we mentioned in https://github.com/apache/datafusion/pull/13736 my only real concern with this addition is that it will make `ColumnStatistics` even bigger (each ScalarValue is quite large already and `ColumnStatistics` are copied a bunch

However, I think the "right" fix for that is to move to using a different statistics representation (e.g. `Arc::ColumnStatistics`) so I don't see this as a blocker